samza-core/src/main/java/org/apache/samza/runtime/LocalContainerRunner.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.samza.metrics.MetricsReporter;
+import org.apache.samza.util.ScalaToJavaUtils;
+import org.apache.samza.util.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * LocalContainerRunner is the local runner for Yarn {@link SamzaContainer}s. It is an intermediate step to
+ * have a local runner for yarn before we consolidate the Yarn container and coordination into a
+ * StreamProcessor. This class will be replaced by the {@link org.apache.samza.processor.StreamProcessor} local runner
+ * once that's done.
+ *
+ * Since we don't have the jobCoordinator implementation in Yarn, the components (jobModel and containerId) are directly
+ * inside the runner.
+ */
+public class LocalContainerRunner extends AbstractApplicationRunner {
+  private static final Logger log = LoggerFactory.getLogger(LocalContainerRunner.class);
+  private final JobModel jobModel;
+  private final Integer containerId;
+
+  public LocalContainerRunner(JobModel jobModel, Integer containerId) {
+    super(jobModel.getConfig());
+    this.jobModel = jobModel;
+    this.containerId = containerId;
+  }
+
+  @Override
+  public void run(StreamApplication streamApp) {
+    JmxServer jmxServer = null;
+    try {
+      jmxServer = new JmxServer();
+      ContainerModel containerModel = jobModel.getContainers().get(containerId);
+
+      SamzaContainer container = SamzaContainer$.MODULE$.apply(
+          containerModel.getContainerId(),
+          containerModel,
+          config,
+          jobModel.maxChangeLogStreamPartitions,
+          SamzaContainer.getLocalityManager(containerId, config),
+          jmxServer,
+          Util.javaMapAsScalaMap(new HashMap<String, MetricsReporter>()),
+          null,
+          this);
+      container.run();
+    } finally {
+      if (jmxServer != null) {
+        jmxServer.stop();
+      }
+    }
+  }
+
+  public static void main(String[] args) {
+    setExceptionHandler(() -> System.exit(1));
+
+    Integer containerId = Integer.valueOf(System.getenv(ShellCommandConfig.ENV_CONTAINER_ID()));
+    log.info(String.format("Got container ID: %d", containerId));
+    String coordinatorUrl = System.getenv(ShellCommandConfig.ENV_COORDINATOR_URL());
+    log.info(String.format("Got coordinator URL: %s", coordinatorUrl));
+    int delay = new Random().nextInt(SamzaContainer.DEFAULT_READ_JOBMODEL_DELAY_MS()) + 1;
+    JobModel jobModel = SamzaContainer.readJobModel(coordinatorUrl, delay);
+    Config config = jobModel.getConfig();
+    JobConfig jobConfig = new JobConfig(config);
+    if (jobConfig.getName().isEmpty()) {
+      throw new SamzaException("can not find the job name");
+    }
+    String jobName = jobConfig.getName().get();
+    String jobId = jobConfig.getJobId().getOrElse(ScalaToJavaUtils.defaultValue("1"));
+    MDC.put("containerName", "samza-container-" + containerId);
+    MDC.put("jobName", jobName);
+    MDC.put("jobId", jobId);
+
+    new LocalContainerRunner(jobModel, containerId).run(null);
```

Comment:
It looks weird to have a null parameter here. Can we use runFromConfig(config)? Or simply run()?